addons/sourcemod/scripting/ConsoleChatManager.sp

```diff
@@ -772,3 +690,215 @@ stock bool ItContainColorcode(char[] szMessage)
+	RemoveColorCodes(sTrimText);
+
+	if (g_bBlockSpam)
+	{
+		int currentTime = GetTime();
+		if (strcmp(sText, g_sLastMessage, true) == 0)
+		{
+			if (g_iLastMessageTime != -1 && ((currentTime - g_iLastMessageTime) <= g_iBlockSpamDelay))
+			{
+				g_sLastMessage = sText;
+				g_iLastMessageTime = currentTime;
+				return;
+			}
+		}
+		g_sLastMessage = sText;
+		g_iLastMessageTime = currentTime;
+	}
+
+	char soundp[255], soundt[255];
+	if (g_bTranslation)
+	{
+		if (kv == INVALID_HANDLE)
+			ReadT();
+
+		if (!KvJumpToKey(kv, sText))
+		{
+			KvJumpToKey(kv, sText, true);
+			KvSetString(kv, "default", sText);
+			KvRewind(kv);
+			KeyValuesToFile(kv, g_sPath);
+			KvJumpToKey(kv, sText);
+		}
+
+		bool blocked = (KvGetNum(kv, "blocked", 0) ? true : false);
+		if (blocked)
+		{
+			KvRewind(kv);
+			return;
+		}
+
+		KvGetString(kv, "sound", soundp, sizeof(soundp), "default");
+		if (strcmp(soundp, "default") == 0)
+			FormatEx(soundt, 255, "common/talk.wav");
+		else
+			FormatEx(soundt, 255, soundp);
+	}
+
+	char sFinalText[1024];
+	char sCountryTag[3];
+	char sIP[26];
+	bool isCountable = IsCountable(sTrimText);
+	bool containsDecimal = StringContainDecimal(sTrimText);
+	bool isCountdown = !containsDecimal && isCountable;
+
+	for (int i = 1 ; i < MaxClients; i++)
+	{
+		if (!IsClientInGame(i) || IsFakeClient(i) || IsClientSourceTV(i))
+			continue;
+
+		if (g_bTranslation)
+		{
+			GetClientIP(i, sIP, sizeof(sIP));
+			GeoipCode2(sIP, sCountryTag);
+			KvGetString(kv, sCountryTag, sText, sizeof(sText), "LANGMISSING");
+
+			if (strcmp(sText, "LANGMISSING") == 0)
+				KvGetString(kv, "default", sText, sizeof(sText));
+		}
+
+		FormatEx(sFinalText, sizeof(sFinalText), "%s", sText);
+
+		if (!g_bRemoveConsoleTag || g_bRemoveConsoleTag && (!ItContainSquarebracket(sText) || !ItContainColorcode(sText)))
+			// Because vscript messages can use custom chat colors, don't add console tag in this case
+			if (!bScript)
+				FormatEx(sFinalText, sizeof(sFinalText), "%s%s", g_sConsoleTag, sText);
+
+		if (isCountable && GetRoundTimeAtTimerEnd() > 0)
+		{
+			float fMinutes = GetRoundTimeAtTimerEnd() / 60.0;
+			int minutes = RoundToFloor(fMinutes);
+			int seconds = GetRoundTimeAtTimerEnd() - minutes * 60;
+			char roundTimeText[32];
+
+			FormatEx(roundTimeText, sizeof(roundTimeText), " {orange}@ %i:%s%i", minutes, (seconds < 10 ? "0" : ""), seconds);
+			FormatEx(sFinalText, sizeof(sFinalText), "%s%s", sFinalText, roundTimeText);
+		}
+
+		CPrintToChat(i, sFinalText);
+
+		// Prepare HUD message
+		if (g_bEnableHud)
+		{
+			if (g_bHudMapSymbols)
+				RemoveDuplicatePrefixAndSuffix(sTrimText);
+
+			RemoveTextInBraces(sTrimText, true, true);
+
+			char szMessage[MAXLENGTH_INPUT + 10];
+			if (g_bHudSymbols)
+				FormatEx(szMessage, sizeof(szMessage), ">> %s <<", sTrimText);
+			else
+				FormatEx(szMessage, sizeof(szMessage), "%s", sTrimText);
+
+			PrepareHudMsg(i, szMessage, isCountdown);
+
+			if (isCountable)
```

Comment:
This used to be `isCountdown` but would break if any chat message contained `.` or `,`. I changed this to `isCountable` since it didn't make sense that line 852, which added the `@ <time>`, already used' isCountable'.